torch/_inductor/ir.py

```diff
@@ -3278,7 +3278,9 @@ def realize_input(cls, x):
             return x
         if isinstance(x, BaseView):
             x.realize()
-            if is_storage_and_layout(x.unwrap_view()):
+            if is_storage_and_layout(x.unwrap_view()) and not isinstance(
+                x.unwrap_view().data, ExternKernelAlloc
+            ):
```

Comment:
According to the analysis in https://github.com/pytorch/pytorch/issues/108754#issuecomment-1751908453. The data layout to bmm before and after changes is actual the same. Note that the stride doesn't matter when the corresponding size is 1: `reinterpret_tensor(buf2, (8, 1, 64), (64, 512, 1))` and `reinterpret_tensor(buf2, (8, 1, 64), (64, 0, 1), 0)` are equivalent. We'd better understand why bmm is slower with the latter but not with the former first. Reverting the change here looks a workaround to me.